homeassistant/components/dynalite/switch.py

```diff
@@ -0,0 +1,34 @@
+"""Support for the Dynalite channels and presets as switches."""
+from homeassistant.components.switch import SwitchDevice
+from homeassistant.core import callback
+
+from .dynalitebase import DynaliteBase, async_setup_entry_base
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Record the async_add_entities function to add them later when received from Dynalite."""
+
+    @callback
+    def switch_from_device(device, bridge):
```

Comment:
Why even this and not pass `DynaliteSwitch` directly ?

Reply:
what do you mean? pass it in the function?
    async_setup_entry_base(
        hass, config_entry, async_add_entities, "switch", DynaliteSwitch
    )

Reply:
done. also in light.py